lib/SIL/SILFunctionType.cpp

```diff
@@ -2805,8 +2805,9 @@ static bool areABICompatibleParamsOrReturns(SILType a, SILType b,
     if (inFunction) {
       auto opaqueTypesSubsituted = aa;
       auto *dc = inFunction->getDeclContext();
-      if (!dc)
-        dc = inFunction->getModule().getSwiftModule();
+      auto *currentModule = inFunction->getModule().getSwiftModule();
+      if (!dc || !dc->isChildContextOf(currentModule))
+        dc = currentModule;
```

Comment:
Would it not be correct to always use currentModule here? I think the SILFunction's dc is only meant for debugging.

Reply:
Yes, it would be correct. The intention of using the current functions current decl context is to be able to optimize the case when you have private declarations in the same sourc file.

Along the way of this micro optimization I introduced the bug that Jordan is fixing here :( .